### PR TITLE
Adds Animated Search Form!

### PR DESCRIPTION
**Pull request by IamManchanda**
Demo => http://codepen.io/IamManchanda/pen/ZKaxrQ

### CODE REVIEW

**Comment by IamManchanda** [comment]
@ncoden You didn't congratulate me .... i just hit up no 200!

**Comment by IamManchanda** [comment]
@rafibomb @kball any updates on review/merge??

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+input.animated-search-form[type=text] {
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: don't need `<input>` here - too specific

Also don't need `type="text"` here - too specific - another use case would be `type="search"`

File: src/building-blocks/animated-search-form/animated-search-form.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<input type="text" name="search" placeholder="Search.." class="animated-search-form">
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.html]: `type="search"`

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  width: 10rem;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: should be a variable:

`$animated-search-form-initial-width: 10rem;`

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  box-sizing: border-box;
+  border: 0.125rem solid #ccc;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: use color variable `$light-gray`

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  box-shadow: 0 0 3.125rem rgba(0,0,0, 0.18);
+  border-radius: 0.25rem;
+  font-size: 1rem;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: i think this is default - omit

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
+input.animated-search-form[type=text] {
+  width: 10rem;
+  box-sizing: border-box;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: i think this is default - omit

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  border: 0.125rem solid #ccc;
+  box-shadow: 0 0 3.125rem rgba(0,0,0, 0.18);
+  border-radius: 0.25rem;
+  font-size: 1rem;
+  background-color: $white;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: default already right?

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  background-image: url('//image.ibb.co/i7NbrQ/search_icon_15.png');
+  background-position: 0.625rem 0.625rem;
+  background-repeat: no-repeat;
+  padding: 0.75rem 1.25rem 0.75rem 2rem;
+  transition: width 0.4s ease-in-out;
+  &:focus {
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: single line break

File: src/building-blocks/animated-search-form/animated-search-form.scss [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  border: 0.125rem solid #ccc;
+  box-shadow: 0 0 3.125rem rgba(0,0,0, 0.18);
+  border-radius: 0.25rem;
+  font-size: 1rem;
+  background-color: $white;
+  background-image: url('//image.ibb.co/i7NbrQ/search_icon_15.png');
+  background-position: 0.625rem 0.625rem;
+  background-repeat: no-repeat;
+  padding: 0.75rem 1.25rem 0.75rem 2rem;
+  transition: width 0.4s ease-in-out;
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.scss]: make this timing a variable

File: src/building-blocks/animated-search-form/animated-search-form.yml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+name: 'Animated Search Form'
+tags:
+  - animation
+  - search
+  - input
+  - form
+category: control
+description: An Animated Search Form that gets full width on focus.
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.yml]: An Animated Search Form that expands to full-width on focus.

File: src/building-blocks/animated-search-form/animated-search-form.yml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+name: 'Animated Search Form'
+tags:
+  - animation
+  - search
+  - input
+  - form
+category: control
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/animated-search-form.yml]: forms

File: src/building-blocks/animated-search-form/layout.html [hunk 12]
```diff
@@ -0,0 +1,3 @@
+<div class="row small-12 medium-8 large-6 columns align-center">
```
Comment (rafibomb) [inline src/building-blocks/animated-search-form/layout.html]: no prob here - you can use the `.containerClass` in yml next time

**Comment by rafibomb** [comment]
Very cool component! Needs a few updates then good for merge!

File: src/building-blocks/animated-search-form/layout.html (hunk 12 above)
Comment (IamManchanda) [inline src/building-blocks/animated-search-form/layout.html]: Think this is also ohk :)

**Comment by IamManchanda** [comment]
done kindly check now @rafibomb @kball !

**Comment by IamManchanda** [comment]
@rafibomb is there anything left in here ??

**Comment by rafibomb** [comment]
Looks good to me!

**Pull request merged**